### ISSUE TITLE
Investigate contrib to handle decorative images

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#8683]
# Story

As a web team, we would like to investigate the contrib space in order to handle alt text for decorative images so we can ensure we follow accessibility best practices and meet compliance.

# Additional Info

Discussed investigating contrib space more before going in our own direction.

https://www.drupal.org/project/decorative_images
https://www.drupal.org/project/decorative_image_widget

# Grooming conversation

https://github.com/uiowa/uiowa/issues/3935#issuecomment-2816173236

Sounds like there is interest in evaluating the contrib space vs doing this ourselves if we can get feedback from TM.

https://www.drupal.org/project/decorative_images - may be redundant with role and empty alt?
https://www.drupal.org/project/decorative_image_widget - want to make sure per instance doesn't get overridden by global default alt text. - https://github.com/uiowa/uiowa/issues/3935#issuecomment-1043188833

Can we address this in all the places we generate programmatic image renders?

Discussed if we did tackle this ourselves there was some interest in keeping alt required and forcing folks to enter an "" or something that we then can strip out.

**Comment by james917** [comment]
Probably worth noting we found the place to adjust at /admin/structure/media/manage/image/fields/media.image.field_media_image

<img width="920" alt="Image" src="https://github.com/user-attachments/assets/2e626237-08a0-4d91-8ef4-c10f53167a40" />

**Comment by bspeare** [comment]
For `decorative_images`, I got this error when trying to upload new media `Error: Call to undefined method Drupal\media_library\Form\FileUploadForm::getEntity() in Drupal\decorative_images\Hook\DecorativeWidgetFormAlter::alterForm() (line 25 of /var/www/html/docroot/modules/contrib/decorative_images/src/Hook/DecorativeWidgetFormAlter.php).`. 

When it did work, here's what the markup looked like:

<img width="905" alt="Image" src="https://github.com/user-attachments/assets/fb58132a-23e8-4e7d-8004-389b10a16bcf" />

**Comment by bspeare** [comment]
For `decorative_image_widget`, you turn it on at `/admin/structure/media/manage/image/form-display`, and `/admin/structure/media/manage/image/form-display/media_browsers`, and `/admin/structure/media/manage/image/form-display/media_library`.   The markup looks like `<img loading="lazy" src="/sites/sandbox.uiowa.edu/files/2025-04/2024_05_06-Cap-and-Gown-Commercial-Shoot-II-jatorner--Cropped_6.jpg" width="2952" height="1516" alt="" class="lazyload loaded" data-once="lazy-load-animation">`. 

Per instance override works just fine. 

`<img loading="lazy" src="/sites/sandbox.uiowa.edu/files/2025-04/2024_05_06-Cap-and-Gown-Commercial-Shoot-II-jatorner--Cropped_6.jpg" width="2952" height="1516" alt="This is the alternative text" class="lazyload loaded" data-once="lazy-load-animation">`

<img width="392" alt="Image" src="https://github.com/user-attachments/assets/e1895751-765c-4372-bef2-d0dfa8539dd2" />

<img width="1071" alt="Image" src="https://github.com/user-attachments/assets/423b414b-5050-449f-9466-1e0d8e79917d" />

I think this module does what we are looking for.  

@joewhitsitt can you provide an example where we "generate programmatic image renders" so that I can check how this module handles that?

**Comment by joewhitsitt** [comment]
It appears our platform has evolved but I could have sworn we were programmatically rendering media images for things like banner video screenshot, featured image in v2, but maybe not.

We do have non-media type stuff like external images (events), aggregator feed image, no-image person, and site specific stuff (search for `<img` in the /sites directory (admissions, lockup)), but I guess that would be out of scope.

**Comment by bspeare** [comment]
Ran WAVE, Siteimprove, and AInspector on the empty alt tags and did not get a violation as they are read as decorative.   WAVE noted the empty tags and gave this explainer:

**Null or empty alternative text**

**What It Means**
Alternative text is null or empty (alt="").

**Why It Matters**
If an image does not convey content or if the content of the image is conveyed elsewhere (such as in a caption or nearby text), the image should have empty/null alternative text (alt="") to ensure that it is ignored by a screen reader and is hidden when images are disabled or unavailable.

**How to Fix It**
Ensure that the image does not convey content or that the content of the image is conveyed in nearby text (e.g., a caption).

**The Algorithm... in English**
An image has alt="".

**Comment by briand44** [comment]
Created a follow up slice, https://github.com/uiowa/uiowa/issues/8711. Closing this as complete!
